segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -205,8 +215,55 @@ private ByteBuffer copyData(List<ReadResultEntryContents> contents) {
         }
         return data;
     }
+    
+    @Override
+    public void updateSegmentAttribute(UpdateSegmentAttribute updateSegmentAttribute) {
+        long requestId = updateSegmentAttribute.getRequestId();
+        String segmentName = updateSegmentAttribute.getSegmentName();
+        UUID attributeId = updateSegmentAttribute.getAttributeId();
+        long newValue = updateSegmentAttribute.getNewValue();
+        long expectedValue = updateSegmentAttribute.getExpectedValue();
+        val update = new AttributeUpdate(attributeId, AttributeUpdateType.ReplaceIfEquals, newValue, expectedValue);
+        segmentStore.updateAttributes(segmentName, Collections.singletonList(update), TIMEOUT).whenComplete((v, e) -> {
+            if (e == null) {
+                connection.send(new SegmentAttributeUpdated(requestId, true));
+            } else {
+                if (ExceptionHelpers.getRealException(e) instanceof BadAttributeUpdateException) {
+                    connection.send(new SegmentAttributeUpdated(requestId, false));
+                } else {
+                    handleException(requestId, segmentName, "Update attribute", e);
+                }
+            }
+        }).exceptionally((Throwable e) -> {
+            handleException(requestId, segmentName, "Update attribute", e);
```

Comment:
Can this be included in whenComplete? Or is it the catch block for the whenComplete arg?